src/wrappers/themis/android/build.gradle

```diff
@@ -62,42 +70,22 @@ android {
     useLibrary 'android.test.base'
     useLibrary 'android.test.mock'
 
-    // TODO(ilammy, 2021-02-21): improve sanity and make this less brittle
-    // This piece of arcane magick ensures that BoringSSL is built before Android stuff.
-    // However, we also need to ensure that BoringSSL artifacts are NOT published
-    // when JavaThemis is published.
-    // There should be a better way to specify build dependencies, right?
-    // Or at least make it a list of depenedencies to add, not to exclude.
-    tasks.whenTaskAdded { task ->
-        def excludeBoringSSL = [
-            'publish',
-            'publishToMavenLocal',
-            'publishProductionPublicationToMavenRepository',
-            'publishProductionPublicationToMavenLocal',
-            'generatePomFileForProductionPublication',
```

Comment:
will this change the way we publish Android Themis?

Reply:
> will this change the way we publish Android Themis?

No. Same tasks for `:android` should be used and should work.

I didn't test tho. Later, since this needs to be tested against that golden Docker image, which is probably royally rotten by now and will need to be updated as well to work with all this new SDK stuff.

This removed block was in fact a hack setting up dependencies between `:android` and `:boringssl` projects, so that BoringSSL binaries are build before Android JNI code is build. That broke down with new NDK since the task names for building CMake and ndk-build are now different. However, with the build unified into CMake, this hack become unnecessary in principle – thus off it goes.